workflow-templates/im-deploy-iis-website.yml

```
@@ -248,6 +161,84 @@ jobs:
+          # to resolve correctly
+          IIS_SERVER@dev: ''
+          IIS_SERVER@qa: ''
+          IIS_SERVER@stage: ''
+          IIS_SERVER@demo: ''
+          IIS_SERVER@uat: ''
+          IIS_SERVER@prod: ''
+          # TODO: Fill in the website-name and remove the environments that are not used.
+          WEBSITE_NAME@dev: ''
+          WEBSITE_NAME@qa: ''
+          WEBSITE_NAME@stage: ''
+          WEBSITE_NAME@demo: ''
+          WEBSITE_NAME@uat: ''
+          WEBSITE_NAME@prod: ''
+          # TODO: Fill in the app-pool-name and remove the environments that are not used.
+          APP_POOL_NAME@dev: ''
+          APP_POOL_NAME@qa: ''
+          APP_POOL_NAME@stage: ''
+          APP_POOL_NAME@demo: ''
+          APP_POOL_NAME@uat: ''
+          APP_POOL_NAME@prod: ''
+          # TODO: Fill in the website-host-header and remove the environments that are not used.
+          WEBSITE_HOST_HEADER@dev: ''
+          WEBSITE_HOST_HEADER@qa: ''
+          WEBSITE_HOST_HEADER@stage: ''
+          WEBSITE_HOST_HEADER@demo: ''
+          WEBSITE_HOST_HEADER@uat: ''
+          WEBSITE_HOST_HEADER@prod: ''
+          # TODO: Fill in the website-path and remove the environments that are not used.
+          WEBSITE_PATH@dev: ''
+          WEBSITE_PATH@qa: ''
+          WEBSITE_PATH@stage: ''
+          WEBSITE_PATH@demo: ''
+          WEBSITE_PATH@uat: ''
+          WEBSITE_PATH@prod: ''
+          # TODO: Fill in the website-cert-path and remove the environments that are not used.
+          WEBSITE_CERT_PATH@dev: ''
+          WEBSITE_CERT_PATH@qa: ''
+          WEBSITE_CERT_PATH@stage: ''
+          WEBSITE_CERT_PATH@demo: ''
+          WEBSITE_CERT_PATH@uat: ''
+          WEBSITE_CERT_PATH@prod: ''
+          # TODO: Fill in the website-cert-friendly-name and remove the environments that are not used.
+          WEBSITE_CERT_FRIENDLY_NAME@dev: ''
+          WEBSITE_CERT_FRIENDLY_NAME@qa: ''
+          WEBSITE_CERT_FRIENDLY_NAME@stage: ''
+          WEBSITE_CERT_FRIENDLY_NAME@demo: ''
+          WEBSITE_CERT_FRIENDLY_NAME@uat: ''
+          WEBSITE_CERT_FRIENDLY_NAME@prod: ''
+          # TODO: Fill in the website-service-account-id and remove the environments that are not used.
+          WEBSITE_SERVICE_ACCOUNT_ID@dev: ''
+          WEBSITE_SERVICE_ACCOUNT_ID@qa: ''
+          WEBSITE_SERVICE_ACCOUNT_ID@stage: ''
+          WEBSITE_SERVICE_ACCOUNT_ID@demo: ''
+          WEBSITE_SERVICE_ACCOUNT_ID@uat: ''
+          WEBSITE_SERVICE_ACCOUNT_ID@prod: ''
+          # TODO: Fill in the website-service-account-secret and remove the environments that are not used.
+          WEBSITE_SERVICE_ACCOUNT_SECRET@dev: ''
+          WEBSITE_SERVICE_ACCOUNT_SECRET@qa: ''
+          WEBSITE_SERVICE_ACCOUNT_SECRET@stage: ''
+          WEBSITE_SERVICE_ACCOUNT_SECRET@demo: ''
+          WEBSITE_SERVICE_ACCOUNT_SECRET@uat: ''
+          WEBSITE_SERVICE_ACCOUNT_SECRET@prod: ''
```

It doesn't look like WEBSITE_SERVICE_ACCOUNT_SECRET is actually used anywhere, but if it was being used it should follow the same suggestions as the other secrets.

I think we should be setting the deployment user ID in this group with the others.
```suggestion
          # TODO: Fill in the DEPLOYMENT_SA_ID and remove the environments that are not used.
          DEPLOYMENT_SA_ID@dev: ''
          DEPLOYMENT_SA_ID@qa: ''
          DEPLOYMENT_SA_ID@stage: ''
          DEPLOYMENT_SA_ID@demo: ''
          DEPLOYMENT_SA_ID@uat: ''
          DEPLOYMENT_SA_ID@prod: ''
```

---

You're right. The service accounts should just be pulled out of secrets, and the `WEBSITE_SERVICE_ACCOUNT_*` should really be for running the app pool and is used in site creation part. I'll fix this and that section together.